src/merkle/store/mod.rs

```diff
@@ -272,20 +278,22 @@ impl MerkleStore {
         Ok(tree.nodes[1])
     }
 
-    /// Adds all the nodes of a Sparse Merkle tree represented by `entries`.
+    /// Adds all the `entries` to a Sparse Merkle tree with `depth`.
     ///
-    /// This will instantiate a Sparse Merkle tree using `entries` and include all the nodes into
-    /// the store.
+    /// All the `entries`, including its paths to the root, will be added to the store.
     ///
     /// # Errors
-    ///
-    /// This will return `InvalidEntriesCount` if the length of `entries` is not `63`.
-    pub fn add_sparse_merkle_tree<R, I>(&mut self, entries: R) -> Result<Word, MerkleError>
+    /// Will return an error if the provided `depth` is greater than [SimpleSmt::MAX_DEPTH].
```

Comment:
I think we should mention the return value here and I'd also rephrase this somewhat. Maybe something like this:
```
Adds a Sparse Merkle tree defined by the specified `entries` to the store and returns the root
of the added tree.

The entries are expected to contain tuples of `(index, node)` describing nodes in the tree at
the depth specified by the `depth` parameter.

# Errors
Returns an error if the provided `depth` is greater than [SimpleSmt::MAX_DEPTH].
```